src/Raven.Server/Documents/Replication/DocumentReplicationLoader.cs

```diff
@@ -728,9 +730,44 @@ public void Dispose()
             foreach (var outgoing in _outgoing)
                 ea.Execute(outgoing.Dispose);
 
+            _database.DocumentTombstoneCleaner?.Unsubscribe(this);
+
             ea.ThrowIfNeeded();
         }
 
+       
+        public Dictionary<string, long> GetLastProcessedDocumentTombstonesPerCollection()
+        {
+            var result = new Dictionary<string, long>(StringComparer.OrdinalIgnoreCase);
+            if (_outgoing.Count == 0)
+            {
+                return result;
+            }
+            // last know etag from source
+            var minEtag = _outgoing.Select(o => o._lastSentDocumentEtag).Min();
```

Comment:
This is wrong. What happens if the remote server has just restarted and isn't online. You are looking only at the _connected_ servers.
Instead, you need to keep track of all of the confirmations that we previously got across all of them.